### ISSUE TITLE
After training the target detection model, the inference results of the test set images show that the labels are opposite

### ISSUE THREAD

**Issue by Leon-Brant** [open-mmlab/mmdetection#8305]
I trained a target detection model using Cascade_mask_RCNn_R50_FPN. My dataset was in COCO format and there were only two kinds of images: Phone and BAG. I made sure there was nothing wrong with my labeling or the way I trained the model. But I found a bug when reasoning with the test set images: the results detected were the opposite! I am using version 2.25 of MMDetection, I would like to ask whether this is the bug of the version that leads to the opposite inference result tag?

Incidentally, after discovering this problem, I tried to use the images of the training set for inference verification, but the inference results still showed opposite labels
![2022-07-05 151810](https://user-images.githubusercontent.com/97673806/177272068-e4c78796-bcc1-4b05-a201-e2bb28f1b8ab.png)
![2022-07-05 151907](https://user-images.githubusercontent.com/97673806/177272091-45bec3d1-a720-41b1-97a8-ec05f7d6728b.png)
![20220705151933](https://user-images.githubusercontent.com/97673806/177272100-b7931531-71d9-43e0-b648-9a17d4e88e8a.png)

**Comment by Leon-Brant** [comment]
> 请提供配置文件'E:/mmdetection_inference/configs/cascade_rcnn/cascade_mask_rcnn_r50_fpn_1x_coco.py'
This is my config file,I sincerely look forward to your reply
[cascade_mask_rcnn_r50_fpn_1x_coco.md](https://github.com/open-mmlab/mmdetection/files/9054744/cascade_mask_rcnn_r50_fpn_1x_coco.md)

**Comment by chhluo** [comment]
You should set values for `classes` when use your own dataset.
```python
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file=...,
        img_prefix=...,
        classes=('bag', 'phone'), # or ('phone', 'bag') ?
)
```

**Comment by Leon-Brant** [comment]
> You should set values for `classes` when use your own dataset.
> 
> ```python
> data = dict(
>     samples_per_gpu=2,
>     workers_per_gpu=2,
>     train=dict(
>         type='CocoDataset',
>         ann_file=...,
>         img_prefix=...,
>         classes=('bag', 'phone'), # or ('phone', 'bag') ?
> )
> ```

I just changed my configuration file **cascade_mask_rcnn_r50_fpn_1x_coco.py** according to the method you said.
But neither adding classes=('bag', 'phone') nor classes=('phone', 'bag') seemed to work, and the trained model still recognized the phone as a bag.
Should I retrain the cascade_mask_rcnn_r50_fpn model and modify my configuration file as you suggested before starting training? Is that what you mean?
<img width="71" alt="image" src="https://user-images.githubusercontent.com/97673806/177668445-44861f85-2d30-4003-ab40-54df23f40150.png">

**Comment by chhluo** [comment]
The following code which set the `model.classes` with the `train_dataset.CLASSES` before training https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/tools/train.py#L230.

See https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/apis/inference.py#L48,
model loads `classes` from checkpoint before inference.

You can directly change the value of `model.classes` for now.

If you want to retain, the following is required
```python
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file=...,
        img_prefix=...,
        classes=('bag', 'phone'), # or ('phone', 'bag') ?
    val =dict(
        classes=('bag', 'phone'), # or ('phone', 'bag') ?
    test =dict(
        classes=('bag', 'phone'), # or ('phone', 'bag') ?
)
```


At the last, you can use tools/misc/browse_dataset.py to check whether the dataset is right.

**Comment by Leon-Brant** [comment]
> 

Thank you very much for your professional reply！In the past time, I have remodified the configuration files and retrained the model as you provided. When infer with the test set images using the retrained model, I finally got the right reasoning, and the labels finally became correct.  As you said, the following changes must be made before the model can be trained with custom data sets：
    1. Change the line 230 of  mmdetection/tools/train.py to model.CLASSES = ['phone', 'bag'] 
    2. Chane the line 48 of  mmdetection/mmdet/apis/inference.py to model.CLASSES = checkpoint['phone', 'bag'] 
    3. Make the following changes to mmdetection\configs\_base_\datasets\coco_instance.py :
        data = dict(
         samples_per_gpu=2,
         workers_per_gpu=2,
        train=dict(
                 type='CocoDataset',
                 ann_file=...,
                  img_prefix=...,
                  classes=('phone', 'bag') 
         val =dict(
                  classes= ('phone', 'bag') 
         test =dict(
                   classes=('phone', 'bag') 
)
  Finally, I got the right result.
<img width="212" alt="image" src="https://user-images.githubusercontent.com/97673806/179898409-f598f936-645d-411f-b3c4-0a0e57f325c0.png">

<img width="109" alt="image" src="https://user-images.githubusercontent.com/97673806/179898497-e16cb017-fe17-4da2-a1e2-e9617158bdce.png">
